overlord/snapstate/autorefresh.go

```diff
@@ -262,6 +267,16 @@ func (m *autoRefresh) canRefreshRespectingMetered(now, lastRefresh time.Time) (c
 	return false, nil
 }
 
+func (m *autoRefresh) waitForNTPSynchronized(maxWait time.Duration) bool {
+	if m.ntpSynchronized || m.startTime.Add(maxWait).Before(time.Now()) {
```

Comment:
shouldn't this be 

```suggestion
	if m.ntpSynchronized || m.startTime.Add(maxWait).After(time.Now()) {
```

since we want to see if the max time has elapsed, we check to see if `start + max > now` ?